### ISSUE TITLE
Valhesia Structures crash

### ISSUE THREAD

**Issue by ghost** [stal111/Valhelsia-Structures#46]
```
Description: Exception in server tick loop

java.util.concurrent.CompletionException: net.minecraft.crash.ReportedException: Exception generating new chunk
	at java.util.concurrent.CompletableFuture.encodeThrowable(Unknown Source) ~[?:1.8.0_261] {}
	at java.util.concurrent.CompletableFuture.completeThrowable(Unknown Source) ~[?:1.8.0_261] {}
	at java.util.concurrent.CompletableFuture.uniCompose(Unknown Source) ~[?:1.8.0_261] {}
	at java.util.concurrent.CompletableFuture$UniCompose.tryFire(Unknown Source) ~[?:1.8.0_261] {}
	at java.util.concurrent.CompletableFuture$Completion.run(Unknown Source) ~[?:1.8.0_261] {}
	at net.minecraft.world.chunk.ChunkTaskPriorityQueueSorter.func_219083_b(SourceFile:58) ~[?:?] {re:classloading}
	at net.minecraft.util.concurrent.DelegatedTaskExecutor.func_213148_e(SourceFile:94) ~[?:?] {re:classloading}
	at net.minecraft.util.concurrent.DelegatedTaskExecutor.func_213145_a(SourceFile:137) ~[?:?] {re:classloading}
	at net.minecraft.util.concurrent.DelegatedTaskExecutor.run(SourceFile:105) ~[?:?] {re:classloading}
	at java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(Unknown Source) ~[?:1.8.0_261] {}
	at java.util.concurrent.ForkJoinTask.doExec(Unknown Source) ~[?:1.8.0_261] {}
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(Unknown Source) ~[?:1.8.0_261] {}
	at java.util.concurrent.ForkJoinPool.runWorker(Unknown Source) ~[?:1.8.0_261] {re:computing_frames}
	at java.util.concurrent.ForkJoinWorkerThread.run(Unknown Source) ~[?:1.8.0_261] {}
Caused by: net.minecraft.crash.ReportedException: Exception generating new chunk
	at net.minecraft.world.server.ChunkManager.lambda$null$18(ChunkManager.java:535) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:A,pl:runtimedistcleaner:A}
	at com.mojang.datafixers.util.Either$Left.map(Either.java:38) ~[?:?] {re:classloading}
	at net.minecraft.world.server.ChunkManager.lambda$chunkGenerate$20(ChunkManager.java:522) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:A,pl:runtimedistcleaner:A}
	... 12 more
Caused by: java.lang.NullPointerException
	at java.util.Objects.requireNonNull(Unknown Source) ~[?:1.8.0_261] {}
	at com.stal111.valhelsia_structures.utils.StructureUtils.checkForOtherStructures(StructureUtils.java:87) ~[valhelsia_structures:16.0.5] {re:classloading}
	at com.stal111.valhelsia_structures.world.structures.AbstractValhelsiaStructure.func_230363_a_(AbstractValhelsiaStructure.java:84) ~[valhelsia_structures:16.0.5] {re:classloading}
	at com.stal111.valhelsia_structures.world.structures.AbstractValhelsiaStructure.func_230363_a_(AbstractValhelsiaStructure.java:44) ~[valhelsia_structures:16.0.5] {re:classloading}
	at net.minecraft.world.gen.feature.structure.Structure.func_242785_a(Structure.java:222) ~[?:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:blame.mixins.json:StructureMixin,pl:mixin:APP:performant.mixins.json:world.StructureMixin,pl:mixin:A}
	at net.minecraft.world.gen.feature.StructureFeature.func_242771_a(SourceFile:35) ~[?:?] {re:mixin,re:classloading}
	at net.minecraft.world.gen.ChunkGenerator.func_242705_a(SourceFile:278) ~[?:?] {re:mixin,re:classloading}
	at net.minecraft.world.gen.ChunkGenerator.func_242707_a(SourceFile:267) ~[?:?] {re:mixin,re:classloading}
	at net.minecraft.world.chunk.ChunkStatus.lambda$static$2(ChunkStatus.java:41) ~[?:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:bettercaves.mixins.json:ChunkStatusMixin,pl:mixin:A}
	at net.minecraft.world.chunk.ChunkStatus.func_223198_a(ChunkStatus.java:198) ~[?:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:bettercaves.mixins.json:ChunkStatusMixin,pl:mixin:A}
	at net.minecraft.world.server.ChunkManager.lambda$null$18(ChunkManager.java:524) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:A,pl:runtimedistcleaner:A}
	at com.mojang.datafixers.util.Either$Left.map(Either.java:38) ~[?:?] {re:classloading}
	at net.minecraft.world.server.ChunkManager.lambda$chunkGenerate$20(ChunkManager.java:522) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:A,pl:runtimedistcleaner:A}
	... 12 more

**Comment by Vaelzan** [comment]
I believe this has been fixed in our Jungle branch (an upcoming update, not out yet). Once that update has released, please comment again if you still have the same issue.

**Comment by sandalle** [comment]
In case it helps for further troubleshooting (if this isn't fixed in the upcoming release), here's a full crash report with valhelsia_structures-16.0.5 on Forge 1.16.5-36.0.21 (also happens with 1.16.5-36.0.26): [crash-2021-02-15_09.45.33-server.txt](https://github.com/ValhelsiaTeam/Valhelsia-Structures/files/5983291/crash-2021-02-15_09.45.33-server.txt)

**Comment by Vaelzan** [comment]
> In case it helps for further troubleshooting (if this isn't fixed in the upcoming release), here's a full crash report with valhelsia_structures-16.0.5 on Forge 1.16.5-36.0.21 (also happens with 1.16.5-36.0.26): [crash-2021-02-15_09.45.33-server.txt](https://github.com/ValhelsiaTeam/Valhelsia-Structures/files/5983291/crash-2021-02-15_09.45.33-server.txt)

You have an entirely different issue - you have an invalid setting in the config file (spacing must always be higher than separation). We've also added clear error messages for the next update, but for now you can prevent the crash by editing the config or deleting it and letting it be recreated.

**Comment by sandalle** [comment]
Oh, sorry. I was doing multiple testing as my world is crashing with a similar error. For that crashlog I had restored my old configs, which I did _not_ manually edit the Valhesia configs. I did one with all brand new configs generated (configs/ , defaultconfigs/, and saves/ were all deleted completely) with the same crash, but just now tried with Valhesia disabled and still got it, so my issue is _not_ from this mod, sorry about the noise. We can delete these posts if you want to clean up the thread. ;)
